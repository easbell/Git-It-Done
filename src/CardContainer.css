.card-container {
  position: relative; 
}

.guess-message {
  position: absolute;
  width: 30rem;
  left: 50%;
  margin-left: -15rem;
  background-color: grey;
  color: purple;
  z-index: 2;
}