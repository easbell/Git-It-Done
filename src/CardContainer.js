import React, { Component } from 'react';
import Card from './Card.js';

export class CardContainer extends Component {
  constructor() {
    super();

    this.state = {
      guessCorrectly: ''
    }
  }

  displayMessage = (message) => {
    message ? this.setState({guessCorrectly: true}) 
            : this.setState({guessCorrectly: false})
    setTimeout(() => {
      this.setState({guessCorrectly: ''})
    }, 1000)
  }
  
  render() {
    return (
      <div className="card-container">
        {
        <div>
          <Card 
            questions={this.props.questions}
            displayMessage={this.displayMessage}
            setToLocalStorage={this.props.setToLocalStorage}
            correctGuesses={this.props.correctGuesses}
          />
        </div>
        }
        <div className="guess-message">
          {this.state.guessCorrectly &&
            <p>You're Right!</p>
          }
          {this.state.guessCorrectly === false &&
            <p>That's wrong!</p>
          }
        </div>
      </div>
    )
  }
}

export default CardContainer
