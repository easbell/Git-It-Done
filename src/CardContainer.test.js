import React from 'react';
import CardContainer from './CardContainer';
import { shallow } from 'enzyme';

const setToLocalStorageMock = jest.fn;
const correctGuessesMock = jest.fn;
const mockData = [
  {
    "id": 1,
    "prompt": "Make a directory",
    "possibleChoices": ["cdir", "mkdir", "newdir"],
    "correctAnswer": "mkdir",
    "save": false
  },
  {
    "id": 2,
    "prompt": "Change directory",
    "possibleChoices": ["change", "cngdir", "cd"],
    "correctAnswer": "cd",
    "save": false
  },
  {
    "id": 3,
    "prompt": "Go to previous directory",
    "possibleChoices": ["cd ..", "cd .", "prev .."],
    "correctAnswer": "cd ..",
    "save": false
  }
]

describe('CardContainer', () => {
  const wrapper = shallow(
    <CardContainer
      allData={mockData}
      setToLocalStorage={setToLocalStorageMock}
      correctGuesses={correctGuessesMock}
    />
  );

  it('should match snapshot with all data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have proper values for default states', () => {
    expect(wrapper.state()).toEqual({guessCorrectly: ''});
  });

  it('should update state when displayMessage is invoked and true is passed in', () => {
    expect(wrapper.state()).toEqual({guessCorrectly: ''});
    wrapper.instance().displayMessage(true);
    expect(wrapper.state()).toEqual({guessCorrectly: true});
  });

  it('should update state when displayMessage is invoked and false is passed in', () => {
    expect(wrapper.state()).toEqual({guessCorrectly: true});
    wrapper.instance().displayMessage(false);
    expect(wrapper.state()).toEqual({guessCorrectly: false});
  });
  
})