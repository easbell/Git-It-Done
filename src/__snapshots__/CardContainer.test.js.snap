// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match snapshot with all data passed in 1`] = `
<div
  className="card-container"
>
  <div>
    <Card
      correctGuesses={[Function]}
      displayMessage={[Function]}
      setToLocalStorage={[Function]}
    />
  </div>
  <div
    className="guess-message"
  />
</div>
`;
